src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,134 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    public static final int DEFAULT_CAPACITY = 10;
+    public static final double DEFAULT_GROW_RATE = 1.5;
+    private Object[] values;
+    private int size;
+
+    public ArrayList() {
+        values = new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
+    private void copyValues(int start, int end, int newStart) {
+        Object[] copy = new Object[end - start + 1];
+        System.arraycopy(values, start, copy, 0, copy.length);
+        System.arraycopy(copy, 0, values, newStart, copy.length);
+    }
+
+    private void checkSizeEnlargeCapacity(int sizeIncrement) {
```

Comment:
it's weird that this checker method has an argument, size == values.length condition is enough and can be calculated by accessing fields directly